crates/vm2/src/instruction_handlers/pointer.rs

```diff
@@ -102,11 +114,31 @@ impl PtrOp for PointerShrink {
     }
 }
 
-use super::monomorphization::*;
+macro_rules! from_ptr_op {
+    ($name:ident <$binop:ty>) => {
+        #[doc = concat!("Creates a [`", stringify!($binop), "`] instruction with the provided params.")]
+        #[inline(always)]
+        pub fn $name(
+            src1: AnySource,
+            src2: Register2,
+            out: AnyDestination,
+            arguments: Arguments,
+            swap: bool,
+        ) -> Self {
+            Self::from_ptr::<$binop>(src1, src2, out, arguments, swap)
+        }
+    };
+}
+
+/// Pointer-related instructions.
+impl<T: Tracer, W: World<T>> Instruction<T, W> {
+    from_ptr_op!(from_pointer_add<PointerAdd>);
+    from_ptr_op!(from_pointer_sub<PointerSub>);
+    from_ptr_op!(from_pointer_pack<PointerPack>);
+    from_ptr_op!(from_pointer_shrink<PointerShrink>);
 
-impl<T: Tracer, W> Instruction<T, W> {
     #[inline(always)]
```

Comment:
The `Instruction::from_*` family sometimes has `#[inline(always)]` and sometimes doesn't. I guess it doesn't hurt because they are only used once but the compiler should be able to figure that out, so I'd say remove all the annotations.